### PR TITLE
Platform dapr integration

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*

Platform dapr integration. to support
- Dapr Implementation to Mgmt and Child clusters
- Component based data store deployments to region specific data store setup
- Resiliency Policies for Micro service to microservice communication
- Resiliency Policies for Micro service to data store communication
- Resiliency Policies such as Retries, TimeOuts, Circuit breakers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml [hunk 1]
```diff
@@ -16,6 +16,10 @@ spec:
   destinations:
```
Comment (shapirov103) [inline platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml:16]: name of the file is appp - typo

File: platform/infra/terraform/mgmt/setups/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+config.yaml
```
Comment (shapirov103) [inline platform/infra/terraform/mgmt/setups/.gitignore:1]: is it by design? I assume we need ability to update config yaml at some point.

File: platform/dapr/apps/node/app.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+//
```
Comment (shapirov103) [inline platform/dapr/apps/node/app.js]: All apps should be under `applications`. I suggest dropping the existing node app and replacing it with this.

Deployments of apps should be under deployment/teams - let's come up with a team name for these and use the existing approach consistently.

File: platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml (hunk 1 above)
Comment (elamaran11) [reply platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml:16]: Its not a typo, it has been an existing file and its named after the CRD `AppProject.argoproj.io` i jused added the `dapr-system` destination to support the deployment.

File: platform/infra/terraform/mgmt/setups/.gitignore (hunk 2 above)
Comment (elamaran11) [reply platform/infra/terraform/mgmt/setups/.gitignore:1]: It is by design and thats correct. You will see there is `default-config.yaml` which holds the default config and `config.yaml` holds the updated deployment one like LB URL. This avoids checking in LB URL accidently.

File: platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml (hunk 1 above)
Comment (shapirov103) [reply platform/infra/terraform/mgmt/packages/argocd/dev/appproject-modern-engg.yaml:16]: Got it, when casing is removed it is unclear.

File: platform/dapr/apps/node/app.js (hunk 3 above)
Comment (elamaran11) [reply platform/dapr/apps/node/app.js]: Fixed

**Review by shapirov103 — COMMENTED** [review]
How are the apps deployed? Do we need to include a simple OAM manifest for these under deployment?

**Comment by elamaran11** [comment]
> How are the apps deployed? Do we need to include a simple OAM manifest for these under deployment?

Apps can be deployed using the YAMLs present in deployment folder for demonstration purposes. Once we start working on implementing DAPR for Apps like Rust, .Net, the sample Node and Python App can be removed.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**